### ISSUE TITLE
DOCS: Check why 'Share' object not linked to from Interest in Schema Reference

### ISSUE THREAD

**Issue by kd-ods** [openownership/data-standard#134]
See /schema/reference.html#interest

(Compare with Source details in schema/reference.html#entitystatement. And check this isn't just a local build issue.)

**Comment by kd-ods** [comment]
This isn't a local build issue.

The difference between Interest > Share and EntityStatement > Source is merely a matter of how they are structured within the (source) schema. Source is housed within [components.json](https://github.com/openownership/data-standard/blob/master/schema/components.json) and $ref'ed from entity-statement.json; Interest is defined in [components.json](https://github.com/openownership/data-standard/blob/master/schema/components.json) and Share is defined within in as a subcomponent.

This is immaterial to the user, though, who would expect both to be treated in the same way in the docs.

If drew on a compiled version of the the schema for the docs then I'd expect that:

* this would resolve itself in that all sub-objects would be treated consistently in the docs
* this resolution would mean that **no** fields would be automatically linked to their sub-objects

Not ideal.

**Comment by odscjames** [comment]
Your right, the current .. jsonschema:: widget decides where to add that "See ___" link solely based on whether it's a ref element or not.

(I can totally see that causing problems in other schemas!)

Need to think about best solution .... and I'm unkeen for that be adding anything to the schema. I'm wondering if we can add something to data-standard/docs/schema/reference.rst so it's like:

```

.. jsonschema:: ../_build_schema/components.json
   :pointer: /definitions/Interest
   :collapse: share,annotations
   :externallinks: interest/share:#share
```

And then the widget looks for the path?

**Comment by kd-ods** [comment]
> the current .. jsonschema:: widget decides where to add that "See ___" link solely based on whether it's a ref element or not.

Hmmm... we can't just add some logic to check for child objects as well as refs?

**Comment by odscjames** [comment]
The problem is currently the widget is *assuming* that if it sees a ref, it can make a link to something in the web browser and that link won't 404. I don't like it assuming anything like that; in fact we have another issue which is exactly about a widget making a link when it shouldn't! I would rather there was some kind of config here.

**Comment by kd-ods** [comment]
>  the widget is assuming that if it sees a ref, it can make a link to something in the web browser and that link won't 404
I agree with you that that is FAR from a safe assumption.

On those grounds, going for the explicit 'externallinks'-type solution seems best. And we need to think about what happens when the relevant field is collapsed and when it isn't.

**Comment by odscjames** [comment]
Have added https://github.com/openownership/data-standard/pull/228 for review.

> And we need to think about what happens when the relevant field is collapsed and when it isn't.

The new external links option will have no effect on fields shown/hidden - if you try and set an external link on a field which is hidden it will just be ignored.

**Comment by kd-ods** [comment]
Fixed in 0.2.
